File: files/CSEC 29.09.R
library(tidyverse)
library(readxl)
library(rebus.datetimes)

path = "files/Excel Challenge September 29th.xlsx"
input = read_excel(path, range = "B2:B6")
test  = read_excel(path, range = "D2:E6")


date_patt = digit(1, 2) %R% "/" %R% digit(1, 2) %R% "/" %R% digit(4,4)
amount_patt = SPACE %R% one_or_more(DIGIT) %R% or(SPACE, END)

result = input %>%
  mutate(Date = str_extract(`Revenues Details`, date_patt) %>% mdy() %>% as.POSIXct(),
         Amount = str_extract(`Revenues Details`, amount_patt) %>% as.numeric()) %>%
  select(-`Revenues Details`)

all.equal(result, test, check.attributes = FALSE)
#> [1] TRUE

File: files/2024-09-29/CSEC 29.09.py
import pandas as pd
import re

path = "files/Excel Challenge September 29th.xlsx"

input = pd.read_excel(path, usecols="B", skiprows=1, nrows=5)
test = pd.read_excel(path, usecols="D:E", skiprows=1, nrows=5)

date_patt = r'\d{1,2}/\d{1,2}/\d{4}'
amount_patt = r'\d+(?=\s|$)'

input['Date'] = input['Revenues Details'].apply(lambda x: pd.to_datetime(re.search(date_patt, x).group(), format='%m/%d/%Y') if re.search(date_patt, x) else None)
input['Amount'] = input['Revenues Details'].apply(lambda x: re.search(amount_patt, x).group().strip() if re.search(amount_patt, x) else None)
input['Amount'] = input['Amount'].astype('float64')

result = input.iloc[:, 1:]

print(result.equals(test)) # True
